### PR TITLE
Simon/unify-tests-interfaces 

### PR DESCRIPTION
**Pull request by SimonRastikian**
This PR is meant to reduce some redundancy in the tests cases and to start allowing generalization of different interfaces.
This tackles a first part of Task https://github.com/near/threshold-signatures/issues/9

The PR tackles two things:
* Simplification of the creation of participants in the tests
* Unification of running DKG functions

A file test.rs was created. It is meant to contain generic testing functions hopefully serving the multiple schemes we have.
Thanks to this file a lot of code is kicked away and replaced with one-liners

### CODE REVIEW

**Comment by gilcu3** [comment]
@SimonRastikian when you get some time I need to know which commits should I keep here when doing rebase. After the parent branch was rebased (or before, when doing force-push) the original commits are obscured so I cannot figure anymore what was added here.

File: src/ecdsa/ot_based_ecdsa/test.rs [hunk 1]
```diff
@@ -68,18 +64,42 @@ pub fn run_presign(
...
+    assert_public_key_invariant(&key_packages);
+    let (pub0, shares0) = deal(&mut OsRng, &participants, threshold).unwrap();
+    let (pub1, shares1) = deal(&mut OsRng, &participants, threshold).unwrap();
+
+    // Presign
+    let mut presign_result = run_presign(key_packages, shares0, shares1, &pub0, &pub1, threshold);
+    presign_result.sort_by_key(|(p, _)| *p);
+
+    let msg = b"hello world";
+    run_sign(
```
Comment (gilcu3) [inline src/ecdsa/ot_based_ecdsa/test.rs:87]: Maybe add an explanatory comment here mentioning that this also checks the signature validity

File: src/test.rs [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+            &new_participants,
+            new_threshold,
+            *p,
+        )?;
+        protocols.push((*p, Box::new(protocol)));
+    }
+
+    let result = run_protocol(protocols)?;
+    Ok(result)
+}
```
Comment (gilcu3) [inline src/test.rs:130]: Afaik this test is checking the case when resharing is done with a new participant. Can we also check the other limits? For example when the number of new participants is the maximum possible, or when some participants leave? My point is we should cover all edge cases.

**Review by gilcu3 — APPROVED** [review]
Great job! 
- I added mainly a comment about covering edges cases for resharing

File: src/ecdsa/ot_based_ecdsa/test.rs (hunk 1 above)
Comment (SimonRastikian) [reply src/ecdsa/ot_based_ecdsa/test.rs:87]: Done! All this is going to be done away with in the upcoming PR so that one has a single signing test function

File: src/test.rs (hunk 2 above)
Comment (SimonRastikian) [reply src/test.rs:130]: This is not true. This function is called when a reshare should happen. Look for example at eddsa/sign.rs where run_reshare is called in both test_reshare_sign_less_participants, and test_reshare_sign_more_participants.

What do you mean when the number of participants is the maximum possible? This would be 2^128 if not more.

Comment (SimonRastikian) [reply src/test.rs:130]: Ok I will open an issue for adding these test cases, and add an extra test case for checking schemes robustness. Maybe also add some cases for malicious signers.

**Comment by SimonRastikian** [comment]
I made a mistake force pushing a wrong rebase. I had to revert the force push by restoring the commits. This is why it seems as if the commits b3bc64e to da8f78e have not been verified yet. But all good. I resolved a small conflict in generate_participants.

**Comment by gilcu3** [comment]
As agreed in https://nearone.slack.com/archives/C0912BTG51T/p1756374506678839?thread_ts=1756368856.422359&cid=C0912BTG51T I will rebase this PR on main.
